src/spring-cloud/azext_spring_cloud/gateway.py

```diff
@@ -0,0 +1,232 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import json
+from sys import version
+from .vendored_sdks.appplatform.v2022_01_01_preview import models
+from ._resource_quantity import validate_cpu, validate_memory
+from knack.util import CLIError
+
+DEFAULT_NAME = "default"
+
+def gateway_update(cmd, client, resource_group, service,
+                   cpu=None,
+                   memory=None,
+                   instance_count=None,
+                   assign_endpoint=None,
+                   https_only=None,
+                   scope=None,
+                   client_id=None,
+                   client_secret=None,
+                   issuer_uri=None,
+                   api_title=None,
+                   api_description=None,
+                   api_documentation_location=None,
+                   api_version=None,
+                   server_url=None,
+                   allowed_origins=None,
+                   allowed_methods=None,
+                   allowed_headers=None,
+                   max_age=None,
+                   allow_credentials=None,
+                   exposed_headers=None
+                   ):
+    cpu = validate_cpu(cpu)
+    memory = validate_memory(memory)
+    all_provided = scope and client_id and client_secret and issuer_uri
+    none_provided = scope is None and client_id is None and client_secret is None and issuer_uri is None
+    if not all_provided and not none_provided :
+        raise CLIError("Single Sign On configurations '--scope --client-id --client-secret --issuer-uri' should be all provided or none provided.")
+
+    gateway = client.gateways.get(resource_group, service, DEFAULT_NAME)
+
+    sso_properties = gateway.properties.sso_properties
+    if all_provided:
+        scope = scope.split(",")
+        sso_properties = models.SsoProperties(
+            scope=scope,
+            client_id=client_id,
+            client_secret=client_secret,
+            issuer_uri=issuer_uri,
+        )
+
+    api_metadata_properties = _update_api_metadata(
+        gateway.properties.api_metadata_properties, api_title, api_description, api_documentation_location, api_version, server_url)
+
+    cors_properties = _update_cors(
```

Comment:
We can group the cors properties to a param group. see https://github.com/Azure/azure-cli/blob/29767d75d850ddc1c24cc85bd46d861b61d77a47/doc/command_guidelines.md#single-complex-parameters

However the nargs uses space sperated which is not the same with what are we using today. We can still putting the splitting method in the validator to split the params. Then the method can accept the valid input